docs/blueprints/linux-setup.md

```diff
@@ -0,0 +1,134 @@
+To generate the deb package, we could try the [dh-make-golang](https://github.com/Debian/dh-make-golang) tool.
+
+The package could then be distributed:
+- by direct download
+- from our own repository
+- in a user repository if it follows the guidelines
+
+When building our package, we can execute  pre-installation and post-installation scripts. We could retrieve the storage node configuration (from command line for instance) from the user using one of these scripts.
+
+#### Agnostic Packaging
+There are [3 major agnostic packaging system](https://www.ostechnix.com/linux-package-managers-compared-appimage-vs-snap-vs-flatpak/) for linux: AppImage, FlatPak and Snap. As AppImage and FlatPak are more desktop application oriented, we choose to focus on Snap.
+
+##### Snap
+[Snaps](https://snapcraft.io/first-snap#go) are containerised software packages. They auto-update daily and work on a variety of Linux distributions. They also revert to the previous version if an update fails. This feature would make it necessary to find out how to implement the rollout versioning.
+
+From the [snap documentation](https://snapcraft.io/docs/go-applications), it seems pretty straightforward to package an application. Snaps are defined in a yaml file. Running an application as a service is done only by specifying "daemon: simple" in the application description. 
+This would make us save the work of building a storage node service.
+
+Snaps can then be published in the snapcraft [app store](https://snapcraft.io/). In the store, we would able to monitor the number of installed snaps. It is possible to [host our own store](https://ubuntu.com/blog/howto-host-your-own-snap-store) but that the snap daemon only handles one repository. Therefore, the use of Canonical's store seems mandatory. Snaps integrate well with [github](https://snapcraft.io/build).
+A snap inside the store can be published in multiple versions in different [channels](https://snapcraft.io/docs/channels).
+
+Snaps have been known for suffering a long start-up time, but it has been [improved](https://snapcraft.io/blog/snap-startup-time-improvements).
+
+#### Comparison between the package and the snap solutions
+Here is a table summarizing the differences between debian packaging system and snaps taken ([source](https://snapcraft.io/blog/a-technical-comparison-between-snaps-and-debs))
+
+| Package	| Debian	| Snap
+| --- | --- | ----
+| Format	| Ar archive	| SquashFS archive
+| Signature verification	| Y (often not used)	| Y
+| Package manager	| dpkg (low-level) Different higher-level  managers available	| snap
+| Front-end	| Many	| Snap Store
+| Installation	| Files copied to /	| Snap uncompressed and mounted as loopback device
+| Dependencies	| Shared	| Inside each snap or content snaps
+| Automatic updates	| semi-automatic	| Y
+| Transactional updates	| N	| Y
+| Multiple installs in parallel	| N	| Y
+| Multiple versions	| N	| Y
+| Security confinement	| Limited	| Y
+| Disk footprint	| Smaller	| Larger
```

Comment:
True in general, because snaps bundle all dependencies, but might not be an issue for us here, because Go likes to compile in all dependencies too.